src/shared/use-data-value-set/use-data-value-set.js

```diff
@@ -45,24 +45,25 @@ function mapDataValuesToFormInitialValues(dataValues) {
     return formInitialValues
 }
 
+const select = (data) => {
```

Comment:
It might be valuable to memoize this function, eg. with `reselect`. This is run on every render.

Reply:
This will happen anyway. A change will optimistically update the response, which causes `data` to change, which in turn will trigger the `select` function.

Reply:
That's not what I mean. The select function is re-run even though the data for the query doesn't change. For every component that is mounted and uses the `useDataValueSet`, or re-renders, the select function is run.

Reply:
See the comment below your suggestion. Let's keep talking there, will resolve this one